src/main/kotlin/com/cognifide/gradle/aem/instance/tasks/InstanceWorkflow.kt

```diff
@@ -0,0 +1,54 @@
+package com.cognifide.gradle.aem.instance.tasks
+
+import com.cognifide.gradle.aem.common.instance.names
+import com.cognifide.gradle.aem.common.tasks.Instance
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+
+open class InstanceWorkflow : Instance() {
+
+    private val notifier = common.notifier
+
+    @get:Internal
+    val workflow by lazy { common.prop.string("instance.workflow") ?: error("No workflow defined!") }
+
+    @get:Internal
+    val workflowPath by lazy { common.prop.string("instance.workflow.path") ?: error("No worflow path defined!") }
+
+    @TaskAction
+    fun run() = anyInstances.forEach { instanceWorkflows(it) }.let { notifier.notify("All workflows executed correctly!", "Instances: ${anyInstances.names}") }
+
+    fun instanceWorkflows(instance: com.cognifide.gradle.aem.common.instance.Instance) {
+        instance.sync.repository {
+            val nodes = sync.repository.query {
+                path(workflowPath)
+                type(NODE_TYPE)
+            }.nodeSequence()
+
+            if (nodes.count() == 0) error("No assets avaialable for processing!")
```

Comment:
Typo

Reply:
And imagine seeing that error when executing some all-in-one task. Please add more context to the message.m that y the error occurred when executing workflow x on instance y for path z etc. Write msgs that you want to see 😎